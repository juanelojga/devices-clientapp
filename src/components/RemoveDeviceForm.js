import React from 'react';
import PropTypes from 'prop-types';

import ConfirmationButton from './ConfirmationButton';

const RemoveDeviceForm = ({ handleClose, handleRemove, loading }) => (
  <div>
    <h3>Are you sure that you want to remove this device?</h3>
    <button onClick={handleClose} disabled={loading}>
      No
    </button>
    <ConfirmationButton
      text="Yes"
      handleClick={handleRemove}
      loading={loading}
      disabled={loading}
    />
  </div>
);

RemoveDeviceForm.propTypes = {
  handleClose: PropTypes.func,
  handleRemove: PropTypes.func,
  loading: PropTypes.bool
};

export default RemoveDeviceForm;
