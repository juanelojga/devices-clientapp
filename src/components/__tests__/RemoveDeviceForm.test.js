import React from 'react';
import { render, fireEvent, cleanup, getByText } from '@testing-library/react';
import 'jest-dom/extend-expect';

import RemoveDeviceForm from '../RemoveDeviceForm';

afterEach(cleanup);

test('snapshot test', () => {
  const handleClose = jest.fn();
  const handleRemove = jest.fn();
  const loading = false;

  const { container } = render(
    <RemoveDeviceForm
      handleClose={handleClose}
      handleRemove={handleRemove}
      loading={loading}
    />
  );
  expect(container).toMatchSnapshot();
});

test('snapshot test loading', () => {
  const handleClose = jest.fn();
  const handleRemove = jest.fn();
  const loading = true;

  const { container } = render(
    <RemoveDeviceForm
      handleClose={handleClose}
      handleRemove={handleRemove}
      loading={loading}
    />
  );
  expect(container).toMatchSnapshot();
});

test('close modal', () => {
  const handleClose = jest.fn();
  const handleRemove = jest.fn();
  const loading = false;

  const { getByText } = render(
    <RemoveDeviceForm
      handleClose={handleClose}
      handleRemove={handleRemove}
      loading={loading}
    />
  );
  fireEvent.click(getByText('No'));
  expect(handleClose).toHaveBeenCalledTimes(1);
  expect(handleRemove).not.toHaveBeenCalled();
});

test('remove device', () => {
  const handleClose = jest.fn();
  const handleRemove = jest.fn();
  const loading = false;

  const { getByText } = render(
    <RemoveDeviceForm
      handleClose={handleClose}
      handleRemove={handleRemove}
      loading={loading}
    />
  );
  fireEvent.click(getByText('Yes'));
  expect(handleRemove).toHaveBeenCalledTimes(1);
  expect(handleClose).not.toHaveBeenCalled();
});

test('disable button if is loading', () => {
  const handleClose = jest.fn();
  const handleRemove = jest.fn();
  const loading = true;

  const { getByText } = render(
    <RemoveDeviceForm
      handleClose={handleClose}
      handleRemove={handleRemove}
      loading={loading}
    />
  );
  const button = getByText('Yes');
  expect(button.disabled).toBeTruthy();
});
