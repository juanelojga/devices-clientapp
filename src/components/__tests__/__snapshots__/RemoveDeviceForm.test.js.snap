// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <div>
    <h3>
      Are you sure that you want to remove this device?
    </h3>
    <button
      class="btn btn-red"
    >
      No
    </button>
      
    <button
      class="btn btn-green"
      type="button"
    >
      Yes
    </button>
  </div>
</div>
`;

exports[`snapshot test loading 1`] = `
<div>
  <div>
    <h3>
      Are you sure that you want to remove this device?
    </h3>
    <button
      class="btn btn-red"
      disabled=""
    >
      No
    </button>
      
    <button
      class="btn btn-green"
      disabled=""
      type="button"
    >
      Yes
      <span
        class="container"
      >
          
          
      </span>
    </button>
  </div>
</div>
`;
